internal/collectors/cpu.go

```diff
@@ -0,0 +1,35 @@
+package collectors
+
+import (
+	"strconv"
+)
+
+type CPU struct {
+}
+
+func (cpu CPU) run(arch Architecture) (Result, error) {
+	switch arch {
+	case ARCHITECTURE_Z:
+		return cpuInfoZ()
+	default:
+		return cpu.cpuInfoDefault()
+	}
+}
+
+func (CPU) cpuInfoDefault() (Result, error) {
+	cpuInfo, err := lscpu()
+	if err != nil {
+		return NoResult, err
+	}
+
+	cpuCount, _ := strconv.Atoi(cpuInfo["CPU(s)"])
+	socket, _ := strconv.Atoi(cpuInfo["Socket(s)"])
+
+	return Result{"cpus": cpuCount, "sockets": socket}, nil
+}
+
+func cpuInfoZ() (Result, error) {
+	return NoResult, nil
+}
+
+// TODO: Set default values for cpu and socket counts to handle invalid output from lscpu(if any) and test it.
```

Comment:
This comment refers to the unused [testdata/collectors/lscpu_x86_64_invalid.txt](https://github.com/SUSE/connect-ng/pull/219/files#diff-ed1033bed86feba30a281c4a2439411fb19536046c07fcb296bce7cd1d8f79aa) it seems and prepares for the situation when cpu count is not detected.